src/util/theme.ts

```diff
@@ -0,0 +1,170 @@
+  switch (color) {
+    case 'red': {
+      return chalk.red(text)
+    }
+
+    case 'blue': {
+      return chalk.blue(text)
+    }
+
+    case 'yellow': {
+      return chalk.yellow(text)
+    }
+
+    case 'green': {
+      return chalk.green(text)
+    }
+
+    case 'cyan': {
+      return chalk.cyan(text)
+    }
+
+    case 'magenta': {
+      return chalk.magenta(text)
+    }
+
+    case 'white': {
+      return chalk.white(text)
+    }
+
+    case 'black': {
+      return chalk.black(text)
+    }
+
+    case 'gray': {
+      return chalk.gray(text)
+    }
+
+    case 'blackBright': {
+      return chalk.blackBright(text)
+    }
+
+    case 'redBright': {
+      return chalk.redBright(text)
+    }
+
+    case 'greenBright': {
+      return chalk.greenBright(text)
+    }
+
+    case 'yellowBright': {
+      return chalk.yellowBright(text)
+    }
+
+    case 'blueBright': {
+      return chalk.blueBright(text)
+    }
+
+    case 'magentaBright': {
+      return chalk.magentaBright(text)
+    }
+
+    case 'cyanBright': {
+      return chalk.cyanBright(text)
+    }
+
+    case 'whiteBright': {
+      return chalk.whiteBright(text)
+    }
+
+    case 'bgBlack': {
+      return chalk.bgBlack(text)
+    }
+
+    case 'bgRed': {
+      return chalk.bgRed(text)
+    }
+
+    case 'bgGreen': {
+      return chalk.bgGreen(text)
+    }
+
+    case 'bgYellow': {
+      return chalk.bgYellow(text)
+    }
+
+    case 'bgBlue': {
+      return chalk.bgBlue(text)
+    }
+
+    case 'bgMagenta': {
+      return chalk.bgMagenta(text)
+    }
+
+    case 'bgCyan': {
+      return chalk.bgCyan(text)
+    }
+
+    case 'bgWhite': {
+      return chalk.bgWhite(text)
+    }
+
+    case 'bgGray': {
+      return chalk.bgGray(text)
+    }
+
+    case 'bgBlackBright': {
+      return chalk.bgBlackBright(text)
+    }
+
+    case 'bgRedBright': {
+      return chalk.bgRedBright(text)
+    }
+
+    case 'bgGreenBright': {
+      return chalk.bgGreenBright(text)
+    }
+
+    case 'bgYellowBright': {
+      return chalk.bgYellowBright(text)
+    }
+
+    case 'bgBlueBright': {
+      return chalk.bgBlueBright(text)
+    }
+
+    case 'bgMagentaBright': {
+      return chalk.bgMagentaBright(text)
+    }
+
+    case 'bgCyanBright': {
+      return chalk.bgCyanBright(text)
+    }
+
+    case 'bgWhiteBright': {
+      return chalk.bgWhiteBright(text)
+    }
+
+    default: {
+      return chalk.gray(text)
+    }
+  }
+}
+
+function isStandardChalk(color: any): color is StandardChalk {
+  return STANDARD_CHALK.includes(color)
+}
+
+export function colorize(color: Color | StandardChalk | undefined, text: string): string {
+  if (isStandardChalk(color)) return standardChalk(color, text)
+
+  return color ? chalk.hex(color.hex())(text) : text
+}
+
+export function parseTheme(untypedTheme: Record<string, string>): Theme {
+  return Object.fromEntries(Object.entries(untypedTheme).map(([key, value]) => [key, getColor(value)]))
```

Comment:
the theme feature should not waste time or memory storing tokens its <b>spec</b> does not support (strict). Its spec supports these tokens:

````js
export const THEME_KEYS = [
  'alias',
  'bin',
  'command',
  'commandSummary',
  'dollarSign',
  'flag',
  'flagDefaultValue',
  'flagOptions',
  'flagRequired',
  'flagSeparator',
  'flagType',
  'sectionDescription',
  'sectionHeader',
  'topic',
  'version',
]
````

Then you have to add these [tests](https://github.com/AllanOricil/core/blob/00a6ccdde08f13b44eec55b57a6d6e584f02d9fd/test/util/util.test.ts#L170-L327) back because they prove the independency and usage of each token of Theme Spec. We need it so that when tokens are changed we know exactly the regressions the change made. For example, if I remove the `alias` token and run the tests, I will see that its test failed, and this will show me a regression:

<img width="1445" alt="image" src="https://github.com/oclif/core/assets/55927613/357ab4c8-4230-4355-b0ce-6462f097440f">

Reply:
@mdonnalley we would change `logToStderr` to the following piece of code so that they can keep using `this.logToStderr` without updating their code. The theme.json is what controls the color changing.

````js
  public static logToStderr(format?: string, ...args: string[]): void {
    write.stderr(colorize(this.config?.theme?.warningMsg, utilFormat(format, ...args) + '\n'))
  }
````

and a new token would be added to the Theme Spec

````js
const THEME_KEYS = {
  ...,
  "warningMsg"
}

````

Reply:
If `this.config?.theme?warningMsg` does not exist, `this.logToStderr` will output what is currently outputing. It would not require any code updates in CLIs using @oclif/core.

Reply:
If CLI developers want to overwrite what the theme.json says, they can do this

````js
export default class MyCommand extends Command {
  public async run(): Promise<void> {
    this.config.theme.warningMsg = getColor('brightBlack')
    this.logToStderr('hello world')
  }
}
````

Reply:
and for `logToStderr`, a better Theme Key is `stderrMessage`,

````js
  public static logToStderr(format?: string, ...args: string[]): void {
    write.stderr(colorize(this.config?.theme?.stderrMessage, utilFormat(format, ...args) + '\n'))
  }
````

Reply:
It's a very poor assumption that everyone would be using `logToStderr` to only log warning messages. But my point was that by allowing any key in `themes.json` we give CLI developers the ability to create a more extensive theme for their users. I have no interest in standardizing themes for every single oclif CLI

Reply:
Going back to what I said earlier, there are no tests to "prove" the Theme Spec the Core Specifies... 

If the core is using certain `keys` everywhere in its code, there must exist tests to prove each single one of these keys....

Reply:
ok, thanks for adding a test back. Now it is working again as I specified.

Reply:
That's what this is: https://github.com/oclif/core/pull/862/files#diff-6bef7aac23bf8d9e00a172402d3b443a7a6e1834198135027ce593fa877e8f73R116